modules/simple-payments/simple-payments.php

```diff
@@ -64,7 +64,11 @@ function register_gutenberg_block() {
 		if ( $this->is_enabled_jetpack_simple_payments() ) {
 			jetpack_register_block( 'jetpack/simple-payments' );
 		} else {
-			Jetpack_Gutenberg::set_extension_unavailable( 'jetpack/simple-payments', 'missing_plan' );
+			$required_plan = 'jetpack_premium';
+			if( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+				$required_plan = 'premium';
+			}
+			Jetpack_Gutenberg::set_extension_unavailable( 'jetpack/simple-payments', $required_plan . '_plan_required' );
```

Comment:
I'm skeptical of putting more information into this string via concatenation.

The plan is still missing, did you consider other alternatives, like adding details?

```php
Jetpack_Gutenberg::set_extension_unavailable(
  'jetpack/simple-payments',
  'missing_plan',
  [
    'required_plan' => ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ? 'premium' : 'jetpack_premium',
  ]
);
```

_interested in discussing the implications of these alternatives, not requesting a change_

Reply:
Yeah, the concatenation is ugly af. What you're suggesting is probably the best way forward (changing `set_extension_unavailable` to add more information). Was a bit hesitant to do it since I've touched those methods sooo many times, at one point cutting back on extra fields :sweat_smile:

Furthermore, the implementation uses an assoc array IIRC (something like `$extensions[ $name ] = $unavailable_reason`), so it'll require some doing. I'll look into it.

Reply:
87d15d8bd